front/lib/models.ts

```diff
@@ -1016,6 +1016,126 @@ User.hasMany(TrackedDocument, {
   onDelete: "CASCADE",
 });
 
+// Events
+export class EventSchema extends Model<
+  InferAttributes<EventSchema>,
+  InferCreationAttributes<EventSchema>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare marker: string;
+  declare description?: string;
+  declare status: "active" | "disabled";
+  declare properties: {
+    name: string;
+    type: number[] | Date[] | string[];
+    description: string;
+  }[];
+  declare userId: ForeignKey<User["id"]>;
+  declare workspaceId: ForeignKey<Workspace["id"]>;
+}
+EventSchema.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    marker: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    description: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+    status: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      defaultValue: "active",
+    },
+    properties: {
+      type: DataTypes.JSONB,
+      allowNull: false,
+    },
+  },
+  {
+    modelName: "event_schema",
+    sequelize: front_sequelize,
+    indexes: [{ fields: ["workspaceId", "marker"], unique: true }],
+  }
+);
+Workspace.hasMany(EventSchema, {
+  foreignKey: { allowNull: false },
+  onDelete: "CASCADE",
+});
+User.hasMany(EventSchema, {
+  foreignKey: { allowNull: false },
+  onDelete: "CASCADE",
+});
+
+export class ExtractedEvent extends Model<
+  InferAttributes<ExtractedEvent>,
+  InferCreationAttributes<ExtractedEvent>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare properties: any;
+
+  declare eventSchemaId: ForeignKey<EventSchema["id"]>;
+  declare documentId: string;
+}
+ExtractedEvent.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    properties: {
+      type: DataTypes.JSONB,
+      allowNull: false,
+    },
+    documentId: {
```

Comment:
Is that enough? I would have expected dataSource + document information?

Reply:
I initially stored the datasourceId but I was not using it so I removed it from the model before submitting this PR. I'm not sure exactly yet how I'm gonna deal with duplicates and I expect having to add it, but I did not want to anticipate something I'm unsure about!

Reply:
Makes sense. It’s just that a document id alone can’t let you retrieve the doc so I would def store it now 👍👍

Reply:
What is the status of this one @PopDaph ? `document_id` is only unique within a given datasource, so I think we definitely need the `dataSourceId` here (and have it non-nullable so might be easier to add now vs once we already have some `ExtractedEvent` rows in the DB)